python/tvm/driver/tvmc/model.py

```diff
@@ -0,0 +1,357 @@
+    """
+
+    def __init__(self, package_path: str):
+        self._tmp_dir = utils.tempdir()
+        self.import_package(package_path)
+
+    def import_package(self, package_path: str):
+        """Load a TVMCPackage from a previously exported TVMCModel.
+
+        Parameters
+        ----------
+        package_path : str
+            The path to the saved TVMCPackage.
+        """
+        temp = self._tmp_dir
+        t = tarfile.open(package_path)
+        t.extractall(temp.relpath("."))
+
+        metadata_path = "metadata.json"
+        with open(temp.relpath(metadata_path), "r") as metadata_file:
+            metadata = json.load(metadata_file)
+        self.name = metadata["model_name"]
+        self.target = metadata["target"]
+        self.runtimes = metadata["runtimes"]
+
+        parameter_path = os.path.join("parameters", f"{self.name}.params")
+        with open(temp.relpath(parameter_path), "rb") as param_file:
+            self.params = bytearray(param_file.read())
+
+        graph_path = os.path.join("runtime-config", "graph", "graph.json")
+        with open(temp.relpath(graph_path), "r") as graph_file:
+            self.graph = graph_file.read()
+
+        ir_path = "relay.txt"
+        with open(temp.relpath(ir_path), "r") as relay_file:
+            self.mod = relay_file.read()
+
+        self.lib_path = temp.relpath(f"{self.name}.so")
+
+    def summary(self):
+        print(self.mod)
+
+
+class TVMCResult(object):
+    """Create a convenience wrapper around the output of tvmc.run
+
+    Parameters
+    ----------
+    outputs : dict
+        Outputs dictionary mapping the name of the output to its numpy value.
+    times : list of str
+        The execution times measured by the time evaluator to produce outputs.
+    """
+
+    def __init__(self, outputs: Dict[str, np.ndarray], times: List[str]):
+        self.outputs = outputs
+        self.times = times
+
+    def format_times(self):
+        """Format the mean, max, min and std of the execution times.
+
+        This has the effect of producing a small table that looks like:
+        .. code-block::
+            Execution time summary:
+            mean (ms)   max (ms)    min (ms)    std (ms)
+            0.14310    0.16161    0.12933    0.01004
+
+        Returns
+        -------
+        str
+            A formatted string containing the statistics.
+        """
+
+        # timestamps
+        mean_ts = np.mean(self.times) * 1000
+        std_ts = np.std(self.times) * 1000
+        max_ts = np.max(self.times) * 1000
+        min_ts = np.min(self.times) * 1000
+
+        header = "Execution time summary:\n{0:^10} {1:^10} {2:^10} {3:^10}".format(
+            "mean (ms)", "max (ms)", "min (ms)", "std (ms)"
+        )
+        stats = "{0:^10.2f} {1:^10.2f} {2:^10.2f} {3:^10.2f}".format(
+            mean_ts, max_ts, min_ts, std_ts
+        )
+
+        return "%s\n%s\n" % (header, stats)
+
+    def get_top_results(self, max_results: int):
```

Comment:
this seems a little specific for the top-level TVMCResult class, and really it should be possible to demonstrate that this class is well-organized by making this a utils function that takes tvmc_result, max_results as parameters. should we move to e.g. `result_util.py`?

Reply:
I disagree, getting the top n results from a neural net is extremely common and seems reasonable to be a method of a TVMCResult. I'd argue against creating more files / code unless there are other utility functions that we think are missing here. If there are quite a few, then I agree having `result_util.py` would be fine.

Reply:
it seems like there is only going to be more introduced, though? I can't see us deleting any. without providing a home for these, then the logical thing to do in follow-on PRs will just be to add them here.

Reply:
its not obvious to me that more will be introduced.